src/Data/Text/Utf8.hs

```diff
@@ -0,0 +1,48 @@
+-- Alfred-Margaret: Fast Aho-Corasick string searching
+-- Copyright 2022 Channable
+--
+-- Licensed under the 3-clause BSD license, see the LICENSE file in the
+-- repository root.
+
+module Data.Text.Utf8 (CodeUnit, CodeUnitIndex(..), unpackUtf8, stringToByteArray, indexTextArray, unicode2utf8) where
+
+
+import           Data.Bits                (Bits, shiftR, (.&.), (.|.))
+import           Data.Char                (ord)
+import           Data.Primitive.ByteArray (ByteArray, byteArrayFromList,
+                                           indexByteArray, sizeofByteArray)
+import           Data.Word                (Word8)
```

Comment:
Nit: we don't align imports in our usual Haskell style because it can result in a lot of diff noise when the longest entry is changed.

Might be a good idea to include a stylish-haskell config from one of our other projects.